.github/workflows/test-all.yml

```
@@ -0,0 +1,25 @@
+name: TestAll
+
+on:
+  pull_request:
+    branches: [ "main", "develop" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
```

Job name "build" should be changed to a "test" or something else.
```suggestion
Integration-test:
```